src/lib/client.ts

```diff
@@ -584,7 +575,8 @@ export default class MqttClient extends TypedEventEmitter<MqttClientEventCallbac
 			options.protocolVersion === 5 && options.customHandleAcks
 				? options.customHandleAcks
 				: (...args) => {
-						args[3](null, 0)
+						// @TODO: This would really need to be made easier to understand. It is difficult to know why the first parameter is Error | number.
+						args[3](0, 0)
```

Comment:
I'm wondering if using 0 instead of null here may lead to some errors elsewhere? I would stick to undefined and fix the type here

Reply:
Good question. This section of the code is treacherous. I think the best course of action would be to refactor it so it is simpler to understand. I am trying to limit the changes I am making to the bare minimum to have a strict TypeScript configuration so the review is easier. I'll see what I can do make this more stable.